### PR TITLE
[RFC] Topology change for multi dai

### PR DESCRIPTION
**Pull request by bardliao**
@plbossart @slawblauciak This is a RFC to explain my idea how multi dai work on kernel side. I verified that sdw_params_stream() will be invoked probably. The only issue I met is that sdw stream didn't support multi cpu yet. ( I haven't tried https://github.com/thesofproject/linux/pull/1802 yet )

### CODE REVIEW

**Comment by bardliao** [comment]
@plbossart I just tested it with #1802. Only rt1308-1 is enabled. I will check what's going on.

**Comment by bardliao** [comment]
> @plbossart I just tested it with #1802. Only rt1308-1 is enabled. I will check what's going on.

Sorry, I forget to update my machine driver. After updating my machine driver, both ALH0x102 and ALH0x202 are enabled. But I will see inconsistent state error on sdw_eneble_stream() with or without #1802. I will look into it. We may still miss something on kernel side.

File: include/sound/sof/dai.h [hunk 1]
```diff
@@ -53,8 +53,15 @@ enum sof_ipc_dai_type {
 	SOF_DAI_INTEL_ALH,		/**< Intel ALH  */
 	SOF_DAI_IMX_SAI,		/**< i.MX SAI */
 	SOF_DAI_IMX_ESAI,		/**< i.MX ESAI */
+	SOF_DAI_MULTIDAI = 0xFFFF       /**< config with multiple elements of the same dai type */
 };
 
+struct sof_ipc_dai_multi_params {
```
Comment (slawblauciak) [inline include/sound/sof/dai.h:59]: Needs a stream map. Also, DAI configurations for the children, unless those are supposed to be passes separately.

**Comment by bardliao** [comment]
Now I can play with multi cpu without error with some hacks. My test branch is https://github.com/bardliao/linux/commits/multi-dai-test

[dapm_on.txt](https://github.com/thesofproject/linux/files/4225668/dapm_on.txt) is the list of dapm widgets which state is On. We can see that two rt1308 amplifiers are on and ALH0x102.OUT is on. I can hear left channel, but not right channel. @slawblauciak Is it possible to copy the data from BUF3 to ALH0x202.OUT? ie. Copy the data from sdw link 1 to sdw link 2. 
I tried connecting ALH0x202 to the same PCM as ALH0x102 on topology.  But I met DSP Panic if I do so. 
After today's experiment, I think we may not need multi dai on FW side. I believe that I can get the same result without multi dai.

**Comment by ranj063** [comment]
@bardliao the PR looks fine. What topology file did you use for your test?

**Comment by plbossart** [comment]
@bardliao I think the discussion for today is that we will go with the multi-cpu DAI, add a demux and check later if we need further changes for pause_push/pause_release cases.
In short we don't need the multi-dai changes for now, so you probably want to close this PR?

**Comment by bardliao** [comment]
Yes, we don't need multi dai :)

**Pull request closed without merging**